### PR TITLE
Log error if clique or ibft have 0 validators in genesis

### PR DESCRIPTION
**Pull request by rain-on**

### CODE REVIEW

File: pantheon/src/main/java/tech/pegasys/pantheon/controller/CliquePantheonControllerBuilder.java [hunk 1]
```diff
@@ -121,6 +126,22 @@ protected MiningCoordinator createMiningCoordinator(
         genesisConfig.getConfigOptions(), nodeKeys, privacyParameters);
   }
 
+  @Override
+  protected boolean validateContext(final ProtocolContext<CliqueContext> context) {
+    final Optional<BlockHeader> genesisBlockHeader =
+        context.getBlockchain().getBlockHeader(GENESIS_BLOCK_NUMBER);
+    if (!genesisBlockHeader.isPresent()) {
+      LOG.error("Genesis block cannot be retriefed from chain.");
```
Comment (ajsutton) [inline pantheon/src/main/java/tech/pegasys/pantheon/controller/CliquePantheonControllerBuilder.java]:
```suggestion
      LOG.error("Genesis block cannot be retrieved from chain.");
```

Comment (ajsutton) [reply pantheon/src/main/java/tech/pegasys/pantheon/controller/CliquePantheonControllerBuilder.java]: I don't think you actually need this check - the blockchain guarantees the genesis block is always available so you could just call `get()` directly.

File: pantheon/src/main/java/tech/pegasys/pantheon/controller/PantheonControllerBuilder.java [hunk 2]
```diff
@@ -299,6 +301,10 @@ protected abstract MiningCoordinator createMiningCoordinator(
 
   protected abstract ProtocolSchedule<C> createProtocolSchedule();
 
+  protected boolean validateContext(final ProtocolContext<C> context) {
+    return true;
```
Comment (ajsutton) [inline pantheon/src/main/java/tech/pegasys/pantheon/controller/PantheonControllerBuilder.java]: Remove the return value - nothing appears to use it.

Comment (rain-on) [reply pantheon/src/main/java/tech/pegasys/pantheon/controller/PantheonControllerBuilder.java]: yeah .... oops .... was thinking that if the genesis was invalid, that may we'd stop execution ... but, think the log is plenty...

**Pull request merged**